packages/common/src/utils/filters.ts

```diff
@@ -779,3 +784,87 @@ export const addDashboardFiltersToMetricQuery = (
         ),
     },
 });
+
+export const createFilterRuleFromRequiredMetricRule = (
+    filter: MetricFilterRule,
+    tableName: string,
+): FilterRule => ({
+    id: filter.id,
+    target: {
+        fieldId: convertFieldRefToFieldId(filter.target.fieldRef, tableName),
+    },
+    operator: filter.operator,
+    values: filter.values,
+    settings: {
+        unitOfTime: filter.settings?.unitOfTime,
+    },
+    required: true,
+});
+
+export const isFilterRuleInQuery = (
+    dimension: Dimension,
+    filterRule: FilterRule,
+    dimensionsFilterGroup: FilterGroup | undefined,
+): undefined | boolean => {
+    let dimensionFieldId = filterRule.target.fieldId;
+    const timeDimension =
+        dimension.isIntervalBase || dimension.timeInterval !== undefined;
+    if (!dimension.isIntervalBase && dimension.timeInterval) {
+        dimensionFieldId = dimensionFieldId.replace(
+            `_${dimension.timeInterval.toLowerCase()}`,
+            '',
+        );
+    }
+    return (
+        dimensionsFilterGroup &&
+        isFilterRuleDefinedForFieldId(
+            dimensionsFilterGroup,
+            dimensionFieldId,
+            timeDimension,
+        )
+    );
+};
+
+export const reduceRequiredDimensionFiltersToFilterRules = (
+    requiredFilters: MetricFilterRule[],
+    table: Table,
+    filters: FilterGroup | undefined,
+): FilterRule[] =>
+    requiredFilters.reduce<FilterRule[]>((acc, filter): FilterRule[] => {
+        const filterRule = createFilterRuleFromRequiredMetricRule(
+            filter,
+            table.name,
+        );
+        const dimension = Object.values(table.dimensions).find(
```

Comment:
nice-to-have 

Wondering if we can remove this find by utilising the keys of `table.dimensions` and getting the field of `filter.target.fieldId` 🤔 but it might not be straightforward - so up to you

Reply:
Hi, gave it a try, but was not clear to me how to make it clean without manipulating the filter.target.fieldId